fuzz/fuzz_targets/deserialize_prefilled_transaction.rs

```diff
@@ -0,0 +1,61 @@
+extern crate bitcoin;
+
+fn do_test(data: &[u8]) {
+    // We already fuzz Transactions in `./deserialize_transaction.rs`.
+    let tx_result: Result<bitcoin::util::bip152::PrefilledTransaction, _> = bitcoin::consensus::encode::deserialize(data);
+
+    match tx_result {
+        Err(_) => {},
+        Ok(mut tx) => {
+            let ser = bitcoin::consensus::encode::serialize(&tx);
+            assert_eq!(&ser[..], data);
+        }
+    }
+}
+
+#[cfg(feature = "afl")]
+#[macro_use] extern crate afl;
+#[cfg(feature = "afl")]
+fn main() {
+    fuzz!(|data| {
+        do_test(&data);
+    });
+}
+
+#[cfg(feature = "honggfuzz")]
+#[macro_use] extern crate honggfuzz;
+#[cfg(feature = "honggfuzz")]
+fn main() {
+    loop {
+        fuzz!(|data| {
+            do_test(data);
+        });
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    fn extend_vec_from_hex(hex: &str, out: &mut Vec<u8>) {
+        let mut b = 0;
+        for (idx, c) in hex.as_bytes().iter().enumerate() {
+            b <<= 4;
+            match *c {
+                b'A'..=b'F' => b |= c - b'A' + 10,
+                b'a'..=b'f' => b |= c - b'a' + 10,
+                b'0'..=b'9' => b |= c - b'0',
+                _ => panic!("Bad hex"),
+            }
+            if (idx & 1) == 1 {
+                out.push(b);
+                b = 0;
+            }
+        }
+    }
+
+    #[test]
+    fn duplicate_crash() {
+        let mut a = Vec::new();
+        extend_vec_from_hex("00000000", &mut a);
```

Comment:
Why not just `super::do_test(&[0; 4])` ?

Reply:
The point is that the user will copy/paste in the hex output from a failing fuzztest.

Reply:
So it's a template? Why not `Vec::from_hex` then?

Reply:
Because this template predates `Vec::from_hex` and we've never thought about it since :)

It would also force all fuzz tests to depend on `bitcoin_hashes`, which may be okay now that we're bringing `hashes` into this repo and unifying all the fuzztests, but in the past it wasn't so reasonable.

Reply:
Well `bitcoin` depends on `bitcoin_hashes` already. Hex will eventually move to a separate crate anyway.